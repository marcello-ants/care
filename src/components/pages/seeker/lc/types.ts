import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
