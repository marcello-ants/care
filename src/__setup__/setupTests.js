import '@testing-library/jest-dom/extend-expect';

jest.mock('@care/analytics', () => {
  // Mock all but init function
  const actual = jest.requireActual('@care/analytics');
  return Object.keys(actual).reduce((acc, method) => {
    if (method === 'init') {
      acc[method] = actual.init;
    } else {
      acc[method] = jest.fn();
    }

    return acc;
  }, {});
});

jest.mock('@/components/hooks/useResizeObserver', () => ({
  __esModule: true,
  default: jest.fn().mockImplementation(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
  })),
}));

jest.mock('@/utilities/utagHelper', () => {
  return {
    TealiumUtagService: { view: () => {} },
  };
});

jest.mock('@/lib/clientLogger');

jest.mock('oidc-client');

jest.mock('next/config', () => () => {
  return {
    publicRuntimeConfig: {
      APP_NAME: 'APP_NAME',
      APP_VERSION: 'APP_VERSION',
      GRAPHQL_URL: 'GRAPHQL_URL',
      OIDC_STS_AUTHORITY: 'OIDC_STS_AUTHORITY',
      OIDC_CLIENT_ID: 'OIDC_CLIENT_ID',
      OIDC_RESPONSE_TYPE: 'OIDC_RESPONSE_TYPE',
      OIDC_CLIENT_SCOPE: 'OIDC_CLIENT_SCOPE',
      OIDC_CALLBACK_PATH: 'OIDC_CALLBACK_PATH',
      OIDC_LOGOUT_URL: 'OIDC_LOGOUT_URL',
      AMPLITUDE_API_KEY: 'AMPLITUDE_API_KEY',
      ANALYTICS_ENABLED: 'ANALYTICS_ENABLED',
      CZEN_GENERAL_LOGIN: 'CZEN_GENERAL_LOGIN',
      CZEN_GENERAL: 'CZEN_GENERAL',
      CZEN_API_KEY: 'CZEN_API_KEY',
      TEALIUM_ENV: 'TEALIUM_ENV',
      SENTRY_ENABLED: 'SENTRY_ENABLED',
      SENTRY_ENV: 'SENTRY_ENV',
      SENTRY_DSN: 'SENTRY_DSN',
      SENTRY_APP_RELEASE: 'SENTRY_APP_RELEASE',
    },
  };
});

beforeEach(() => {
  if (typeof window !== 'undefined') {
    window.localStorage?.clear();
    window.sessionStorage?.clear();
  }
});

// eslint-disable-next-line no-global-assign
global.fetch = jest.fn(() => Promise.resolve({}));

if (typeof document !== 'undefined') {
  const fbScript = document.createElement('script');
  fbScript.id = 'facebook-jssdk';
  document.body.appendChild(fbScript);
}
