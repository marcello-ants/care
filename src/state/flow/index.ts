export type { FlowAction, FlowState } from '../../types/flow';
export { initialState, reducer } from './reducer';
