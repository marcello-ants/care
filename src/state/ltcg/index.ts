export type { LtcgAction, LtcgState } from '@/types/ltcg';
export { initialState, reducer } from './reducer';
