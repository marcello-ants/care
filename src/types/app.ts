import { NextComponentType } from 'next';
import { FC, ReactElement } from 'react';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import { SeekerAction, SeekerState } from './seeker';
import { SeekerCCAction, SeekerCCState } from './seekerCC';
import { SeekerHKAction, SeekerHKState } from './seekerHK';
import { SeekerPCAction, SeekerPCState } from './seekerPC';
import { SeekerTUAction, SeekerTUState } from './seekerTU';
import { ProviderAction, ProviderState } from './provider';
import { ProviderCCAction, ProviderCCState } from './providerCC';
import { FlowAction, FlowState } from './flow';
import { LtcgAction, LtcgState } from './ltcg';
import { JobAction } from './job';

import { ILayoutProps } from './layout';

export interface AppState {
  flow: FlowState;
  seeker: SeekerState;
  seekerCC: SeekerCCState;
  seekerHK: SeekerHKState;
  seekerPC: SeekerPCState;
  seekerTU: SeekerTUState;
  provider: ProviderState;
  providerCC: ProviderCCState;
  ltcg: LtcgState;
  version: string;
}

export type AppAction =
  | SeekerAction
  | SeekerCCAction
  | ProviderAction
  | FlowAction
  | ProviderCCAction
  | SeekerHKAction
  | JobAction
  | SeekerPCAction
  | SeekerTUAction
  | LtcgAction;

export interface AppDispatch {
  (action: AppAction): void;
}

export type AppPageComponent = NextComponentType & {
  // whether the page requires an auth check (defaults to false)
  CheckAuthCookie?: boolean;
  // custom page footer
  Footer?: ReactElement;
  // custom page layout (otherwise uses DefaultLayout)
  Layout?: FC<ILayoutProps>;
  // custom page header
  Header?: ReactElement;
  // tags that should be added to the page's <head>
  HeadTags?: ReactElement;
  // whether the page should use the default page transition (defaults to true)
  usePageTransition?: boolean;
  // when not using the default page transition, the key to use to know when to reenable the page transition (defaults to the route name)
  transitionKey?: string;
  // custom background image
  backgroundImage?: (props?: any) => ClassNameMap<'background'>;
  // whether the automatic "Screen Viewed" amplitude event logging should be disabled
  disableScreenViewed?: boolean;
};
