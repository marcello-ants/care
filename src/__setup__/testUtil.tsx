/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { NextRouter } from 'next/router';
import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import DayJSUtils from '@date-io/dayjs';
import { theme } from '@care/material-ui-theme';
import { ThemeProvider } from '@material-ui/core';
import { FeatureFlags, FeatureFlagsProvider } from '@/components/FeatureFlagsContext';
import { AppStateProvider } from '@/components/AppState';
import { initialAppState } from '@/state';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { AppState } from '@/types/app';

export interface PrepareTreeOptions {
  pathname?: string;
  mocks?: MockedResponse[];
  flags?: FeatureFlags;
  appState?: AppState;
  addTypeName?: boolean;
  basePath?: string;
}

const prepareTree = (options?: PrepareTreeOptions) => {
  const {
    pathname = '/path',
    mocks = [],
    flags = {},
    appState = initialAppState,
    addTypeName = false,
    basePath = '/',
  } = options || {};
  const routerMock = {
    push: jest.fn(),
    asPath: pathname,
    beforePopState: (cb: any) => {
      cb({ url: pathname });
    },
    basePath,
    pathname,
    query: {},
    back: jest.fn(),
  } as unknown as NextRouter;

  const renderFn = (children: React.ReactNode) => (
    <ThemeProvider theme={theme}>
      <RouterContext.Provider value={routerMock}>
        <MockedProvider mocks={mocks} addTypename={addTypeName}>
          <MuiPickersUtilsProvider utils={DayJSUtils}>
            <FeatureFlagsProvider flags={flags}>
              <AppStateProvider initialStateOverride={appState}>{children}</AppStateProvider>
            </FeatureFlagsProvider>
          </MuiPickersUtilsProvider>
        </MockedProvider>
      </RouterContext.Provider>
    </ThemeProvider>
  );

  return {
    renderFn,
    routerMock,
  };
};

export const preRenderHook = (options?: PrepareTreeOptions) => {
  const tree = prepareTree(options);
  const renderFn = <T,>(hook: () => T) =>
    renderHook(hook, {
      wrapper({ children }: { children: React.ReactNode }) {
        return tree.renderFn(children);
      },
    });

  return {
    ...tree,
    renderFn,
  };
};

export const preRenderPage = (options?: PrepareTreeOptions) => {
  const tree = prepareTree(options);
  const renderFn = (ComponentOrChildren: React.ComponentType | React.ReactNode) => {
    if (React.isValidElement(ComponentOrChildren)) {
      return render(tree.renderFn(ComponentOrChildren));
    }

    const Component = ComponentOrChildren as React.ComponentType;
    return render(tree.renderFn(<Component />));
  };

  return {
    ...tree,
    renderFn,
    renderTree: tree.renderFn,
  };
};

export const setupWindowLocation = () => {
  const originalLocation = window.location;
  const windowLocationMock = jest.fn();

  // @ts-ignore
  delete window.location;
  // @ts-ignore
  window.location = {
    assign: windowLocationMock,
  };

  return {
    mock: windowLocationMock,
    cleanUp() {
      window.location = originalLocation;
    },
  };
};

export const setupLocalStorage = () => {
  const originalLocalStorage: Storage = window.localStorage;
  const localStorageMock = jest.fn();
  Object.defineProperty(window, 'localStorage', localStorageMock);

  return {
    mock: localStorageMock,
    cleanUp() {
      Object.defineProperty(window, 'localStorage', originalLocalStorage);
    },
  };
};

export function mockFetch({ ok, status, data }: { ok: boolean; status: number; data: any }) {
  return () =>
    Promise.resolve({
      ok,
      status,
      json: () => Promise.resolve(data),
    } as Response);
}
