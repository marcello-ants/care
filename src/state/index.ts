/* eslint no-param-reassign: ["error", { "props": true, "ignorePropertyModificationsFor": ["draft"] }] */
import produce, { enableES5 } from 'immer';
import { CURRENT_STATE_VERSION, CURRENT_STATE_MAJOR_VERSION } from '../constants';
import { AppAction, AppState } from '../types/app';

import {
  SeekerAction,
  initialState as seekerInitialState,
  reducer as seekerReducer,
  validate as validateSeeker,
  doesSeekerStateHaveDefaultKeys,
} from './seeker';

import {
  SeekerCCAction,
  initialState as seekerCCInitialState,
  reducer as seekerCCReducer,
  validate as validateSeekerCC,
  doesSeekerCCStateHaveDefaultKeys,
} from './seekerCC';

import {
  ProviderAction,
  initialState as providerInitialState,
  reducer as providerReducer,
} from './provider';

import {
  ProviderCCAction,
  initialState as providerCCInitialState,
  reducer as providerCCReducer,
} from './providerCC';

import {
  SeekerHKAction,
  initialState as seekerHKInitialState,
  reducer as seekerHKReducer,
} from './seekerHK';

import {
  SeekerPCAction,
  initialState as seekerPCInitialState,
  reducer as seekerPCReducer,
} from './seekerPC';

import {
  SeekerTUAction,
  initialState as seekerTUInitialState,
  reducer as seekerTUReducer,
} from './seekerTU';

import { FlowAction, initialState as flowInitialState, reducer as flowReducer } from './flow';

import { LtcgAction, initialState as ltcgInitialState, reducer as ltcgReducer } from './ltcg';

// immer uses Proxy, but if that's not available the ES5 plugin needs to be enabled
/* istanbul ignore if */
if (typeof Proxy === 'undefined') {
  // thanks, IE 11
  enableES5();
}

export const initialAppState: AppState = {
  flow: flowInitialState,
  seeker: seekerInitialState,
  seekerCC: seekerCCInitialState,
  provider: providerInitialState,
  providerCC: providerCCInitialState,
  seekerHK: seekerHKInitialState,
  seekerPC: seekerPCInitialState,
  seekerTU: seekerTUInitialState,
  ltcg: ltcgInitialState,
  version: CURRENT_STATE_VERSION,
};

export const rootReducer = produce((draft: AppState, action: AppAction) => {
  draft.seeker = seekerReducer(draft.seeker, action as SeekerAction);
  draft.provider = providerReducer(draft.provider, action as ProviderAction);
  draft.providerCC = providerCCReducer(draft.providerCC, action as ProviderCCAction);
  draft.flow = flowReducer(draft.flow, action as FlowAction);
  draft.seekerCC = seekerCCReducer(draft.seekerCC, action as SeekerCCAction);
  draft.seekerHK = seekerHKReducer(draft.seekerHK, action as SeekerHKAction);
  draft.seekerPC = seekerPCReducer(draft.seekerPC, action as SeekerPCAction);
  draft.seekerTU = seekerTUReducer(draft.seekerTU, action as SeekerTUAction);
  draft.ltcg = ltcgReducer(draft.ltcg, action as LtcgAction);

  return draft;
});

export function validateState(state: AppState, pathName: string) {
  // TODO: call jobPost validation here instead of in AppState
  return validateSeeker(state, pathName) && validateSeekerCC(state, pathName);
}

export function isStateVersionCompatible(state: AppState) {
  if (state?.version) {
    const versionComponents = state.version.split('.');
    if (versionComponents.length === 3 && versionComponents[0] === CURRENT_STATE_MAJOR_VERSION) {
      return true;
    }
  }
  return false;
}

export function doesStateHaveDefaultKeys(state: AppState) {
  return (
    doesSeekerStateHaveDefaultKeys(state.seeker) && doesSeekerCCStateHaveDefaultKeys(state.seekerCC)
  );
}
