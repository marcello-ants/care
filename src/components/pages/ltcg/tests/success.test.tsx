import { screen, waitFor } from '@testing-library/react';
import { GraphQLError } from 'graphql';
import {
  CREATE_HOME_PAY_PROSPECT,
  HIRED_CAREGIVER_CRM_EVENT_CREATE,
} from '@/components/request/GQL';
import logger from '@/lib/clientLogger';
import { initialAppState } from '@/state';
import { AppState } from '@/types/app';
import { preRenderPage } from '@/__setup__/testUtil';
import {
  YesOrNoAnswer,
  HomePayExpectedHireTimeFrame,
  HomepayCaregiverType,
  CareDate,
  ServiceType,
} from '@/__generated__/globalTypes';
import {
  createHomePayProspect,
  createHomePayProspectVariables,
} from '@/__generated__/createHomePayProspect';
import {
  CzenCrmEventHiredCaregiverCreate,
  CzenCrmEventHiredCaregiverCreateVariables,
} from '@/__generated__/CzenCrmEventHiredCaregiverCreate';

import Success from '../success';

const czenCrmEventHiredCaregiverCreateMock = {
  request: {
    query: HIRED_CAREGIVER_CRM_EVENT_CREATE,
    variables: {
      input: {
        czenCrmEventHiredCaregiver: {
          isEnrollment: true,
          vertical: ServiceType.SENIOR_CARE,
        },
        userId: '123456',
      },
    } as CzenCrmEventHiredCaregiverCreateVariables,
  },
  result: {
    data: {
      czenCrmEventHiredCaregiverCreate: {
        __typename: 'CzenCrmEventResponse',
        success: true,
      },
    } as CzenCrmEventHiredCaregiverCreate,
  },
};

const czenCrmEventHiredCaregiverCreateErrorMock = {
  request: {
    query: HIRED_CAREGIVER_CRM_EVENT_CREATE,
    variables: {
      input: {
        czenCrmEventHiredCaregiver: {
          isEnrollment: true,
          vertical: ServiceType.SENIOR_CARE,
        },
        userId: '123456',
      },
    } as CzenCrmEventHiredCaregiverCreateVariables,
  },
  result: {
    data: null,
    errors: [
      {
        message: 'server error',
        locations: [
          {
            line: 2,
            column: 3,
          },
        ],
      },
    ] as unknown as GraphQLError[],
  },
};

const createHomePayProspectMock = {
  request: {
    query: CREATE_HOME_PAY_PROSPECT,
    variables: {
      input: {
        address: {
          addressLine1: '5th street',
          city: 'New York',
          state: 'NY',
          zip: '10001',
        },
        caregiverType: HomepayCaregiverType.SENIOR_CARE,
        email: 'tim@care.com',
        firstName: 'Jon',
        lastName: 'Doe',
        phoneNumber: '2025551313',
        referringSite: 'LTCG',
        hasHired: true,
        planId: 'LtcgNoBilling',
        partnerId: '485',
        expectedHireTimeFrame: HomePayExpectedHireTimeFrame.AS_SOON_AS_POSSIBLE,
      },
    } as createHomePayProspectVariables,
  },
  result: {
    data: {
      createHomePayProspect: {
        __typename: 'CreateHomePayProspectSuccess',
        prospect: {
          prospectId: 'prospectId',
        },
      },
    } as createHomePayProspect,
  },
};

const createHomePayProspectHasNotHiredMock = {
  request: {
    query: CREATE_HOME_PAY_PROSPECT,
    variables: {
      input: {
        address: {
          addressLine1: '5th street',
          city: 'New York',
          state: 'NY',
          zip: '10001',
        },
        caregiverType: HomepayCaregiverType.SENIOR_CARE,
        email: 'tim@care.com',
        firstName: 'Jon',
        lastName: 'Doe',
        phoneNumber: '2025551313',
        referringSite: 'LTCG',
        hasHired: false,
        planId: 'LtcgNoBilling',
        partnerId: '485',
        expectedHireTimeFrame: HomePayExpectedHireTimeFrame.AS_SOON_AS_POSSIBLE,
      },
    } as createHomePayProspectVariables,
  },
  result: {
    data: {
      createHomePayProspect: {
        __typename: 'CreateHomePayProspectSuccess',
        prospect: {
          prospectId: 'prospectId',
        },
      },
    } as createHomePayProspect,
  },
};

const createHomePayProspectErrorMock = {
  request: {
    query: CREATE_HOME_PAY_PROSPECT,
    variables: {
      input: {
        address: {
          addressLine1: '5th street',
          city: 'New York',
          state: 'NY',
          zip: '10001',
        },
        caregiverType: HomepayCaregiverType.SENIOR_CARE,
        email: 'tim@care.com',
        firstName: 'Dwight',
        lastName: 'Schrute',
        phoneNumber: '2025551313',
        referringSite: 'LTCG',
        hasHired: true,
        planId: 'LtcgNoBilling',
        partnerId: '485',
        expectedHireTimeFrame: HomePayExpectedHireTimeFrame.AS_SOON_AS_POSSIBLE,
      },
    } as createHomePayProspectVariables,
  },
  result: {
    data: {
      createHomePayProspect: {
        __typename: 'CreateHomePayProspectError',
        errors: [
          {
            message: 'Error',
          },
        ],
      },
    } as createHomePayProspect,
  },
};

const ltcgState: AppState = {
  ...initialAppState,
  ltcg: {
    ...initialAppState.ltcg,
    careDate: CareDate.RIGHT_NOW,
    caregiverNeeded: YesOrNoAnswer.NO,
    location: {
      city: 'New York',
      state: 'NY',
      zipcode: '10001',
    },
    homePayProspect: {
      address: '5th street',
      email: 'tim@care.com',
      firstName: 'Jon',
      lastName: 'Doe',
      phoneNumber: '+12025551313',
    },
  },
  flow: {
    ...initialAppState.flow,
    memberId: '123456',
  },
};

const getByChildText = () => {
  const firstNode = screen.getByText((content) => content.startsWith('Expect a call from'));
  if (!firstNode) return "Couldn't find node with text starting with 'Expect a call from'.";

  const parent = firstNode.parentElement;
  if (!parent) return 'Could not find parent node.';

  const children = parent.childNodes;
  const stringBuilder: string[] = [];
  children.forEach(({ textContent }) => textContent && stringBuilder.push(textContent));

  // Replace &nbsp; with actual spaces for testing purposes
  return stringBuilder.join('').replace(new RegExp('\u00a0', 'g'), ' ');
};

describe('Success', () => {
  const realLoggerInfo = logger.info;
  const realLoggerError = logger.error;
  const loggerInfoMock = jest.fn();
  const loggerErrorMock = jest.fn();

  beforeAll(() => {
    logger.info = loggerInfoMock;
    logger.error = loggerErrorMock;
  });

  afterEach(() => {
    loggerInfoMock.mockReset();
    loggerErrorMock.mockReset();
  });

  afterAll(() => {
    logger.info = realLoggerInfo;
    logger.error = realLoggerError;
  });
  it('matches snapshot', () => {
    const { renderFn } = preRenderPage({ appState: ltcgState });
    const { asFragment } = renderFn(Success);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the default list', () => {
    const { renderFn } = preRenderPage({ appState: ltcgState });
    renderFn(Success);

    expect(screen.getByText('Reviewing how reimbursement works.')).toBeInTheDocument();
    expect(
      screen.getByText('Setting up payroll between you and your caregiver.')
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        'Ensuring your caregiver has the right tools to capture needed reimbursement information.'
      )
    ).toBeInTheDocument();

    expect(screen.getByText('Be on the lookout for a call from us.')).toBeInTheDocument();
    expect(getByChildText()).toBe(
      ' Expect a call from our team within 1 business day. They are ready to help you learn more about the following:'
    );
  });

  it('should render the looking for caregivers list', () => {
    const customLtcgState = {
      ...ltcgState,
      ltcg: { ...ltcgState.ltcg, caregiverNeeded: YesOrNoAnswer.YES },
    };

    const { renderFn } = preRenderPage({ appState: customLtcgState });
    renderFn(Success);

    expect(screen.getByText('Assessing your needs.')).toBeInTheDocument();
    expect(screen.getByText('Finding great caregivers near you.')).toBeInTheDocument();
    expect(screen.getByText('Working with you to hire the best option.')).toBeInTheDocument();
    expect(screen.getByText('Want to speak with an expert now?')).toBeInTheDocument();
    expect(getByChildText()).toBe(
      ' Expect a call from our team within 1 business day. They are ready to help by doing the following:'
    );
  });

  it('should log a success event for the createHomePayProspect mutation', async () => {
    const { renderFn } = preRenderPage({ appState: ltcgState, mocks: [createHomePayProspectMock] });
    renderFn(Success);

    await waitFor(() =>
      expect(loggerInfoMock).toHaveBeenCalledWith({
        event: 'createHomePayProspectMutationSuccess',
        prospectId: 'prospectId',
      })
    );
  });

  it('should log an error for the createHomePayProspect mutation', async () => {
    const errorLtcgState: AppState = {
      ...initialAppState,
      ltcg: {
        ...initialAppState.ltcg,
        careDate: CareDate.RIGHT_NOW,
        caregiverNeeded: YesOrNoAnswer.NO,
        location: {
          city: 'New York',
          state: 'NY',
          zipcode: '10001',
        },
        homePayProspect: {
          address: '5th street',
          email: 'tim@care.com',
          firstName: 'Dwight',
          lastName: 'Schrute',
          phoneNumber: '+12025551313',
        },
      },
    };

    const { renderFn } = preRenderPage({
      appState: errorLtcgState,
      mocks: [createHomePayProspectErrorMock],
    });
    renderFn(Success);

    await waitFor(() =>
      expect(loggerErrorMock).toHaveBeenCalledWith({
        event: 'createHomePayProspectMutationError',
      })
    );
  });

  it('should log a success event for the czenCrmEventHiredCaregiverCreate mutation when caregiver has been hired', async () => {
    const { renderFn } = preRenderPage({
      appState: ltcgState,
      mocks: [czenCrmEventHiredCaregiverCreateMock, createHomePayProspectMock],
    });
    await renderFn(Success);

    await waitFor(
      () =>
        expect(loggerInfoMock).toHaveBeenCalledWith({
          event: 'createHomePayProspectMutationSuccess',
          prospectId: 'prospectId',
        }),
      { timeout: 1000 }
    );

    expect(loggerInfoMock).toHaveBeenCalledWith({
      event: 'CzenCrmEventHiredCaregiverCreateSuccess',
    });

    expect(loggerErrorMock).toHaveBeenCalledTimes(0);

    expect(loggerInfoMock).toHaveBeenCalledTimes(2);
  });

  it('should not call czenCrmEventHiredCaregiverCreate mutation when caregiver has not been hired', async () => {
    const { renderFn } = preRenderPage({
      appState: {
        ...ltcgState,
        ltcg: {
          ...ltcgState.ltcg,
          careDate: CareDate.RIGHT_NOW,
          caregiverNeeded: YesOrNoAnswer.YES,
        },
      },
      mocks: [createHomePayProspectHasNotHiredMock],
    });
    renderFn(Success);

    await waitFor(
      () =>
        expect(loggerInfoMock).toHaveBeenCalledWith({
          event: 'createHomePayProspectMutationSuccess',
          prospectId: 'prospectId',
        }),
      { timeout: 1000 }
    );
  });

  it('should log error when czenCrmEventHiredCaregiverCreate mutation throws', async () => {
    const { renderFn } = preRenderPage({
      appState: ltcgState,
      mocks: [createHomePayProspectMock, czenCrmEventHiredCaregiverCreateErrorMock],
    });
    renderFn(Success);

    await waitFor(
      () =>
        expect(loggerInfoMock).toHaveBeenCalledWith({
          event: 'createHomePayProspectMutationSuccess',
          prospectId: 'prospectId',
        }),
      { timeout: 1000 }
    );
    expect(loggerErrorMock).toHaveBeenCalledWith({
      error: new Error('server error'),
      event: 'CzenCrmEventHiredCaregiverCreateError',
    });

    expect(loggerErrorMock).toHaveBeenCalledTimes(1);

    expect(loggerInfoMock).toHaveBeenCalledTimes(1);
  });
});
