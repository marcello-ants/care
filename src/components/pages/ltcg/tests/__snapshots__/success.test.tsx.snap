// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-animatedImage-5"
    >
      <img
        alt="Success animation"
        class="makeStyles-animationImage-6"
        src="/app/enrollment/success-animation-2x.gif"
      />
    </div>
    <h2
      class="MuiTypography-root makeStyles-header-11 MuiTypography-h2"
    >
      Thanks for sharing your information! Here's what comes next:
    </h2>
    <p
      class="MuiTypography-root makeStyles-subheader-1 MuiTypography-body2"
    >
       
      <span>
        Expect a call from our team within 1 business day. They are ready to help
      </span>
      <span>
         you learn more about
      </span>
      <span>
        Â the following:
      </span>
    </p>
    <div
      class="makeStyles-list-2"
    >
      <div
        class="makeStyles-listItem-3"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-root-12 makeStyles-listIcon-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.27 16.83a1 1 0 01-.7-.3l-4.3-4.3a1 1 0 011.41-1.41l3.59 3.59 6.8-6.79A1 1 0 0118.48 9L11 16.53a1 1 0 01-.73.3z"
            fill="#172c42"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Reviewing how reimbursement works.
        </p>
      </div>
      <div
        class="makeStyles-listItem-3"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-root-13 makeStyles-listIcon-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.27 16.83a1 1 0 01-.7-.3l-4.3-4.3a1 1 0 011.41-1.41l3.59 3.59 6.8-6.79A1 1 0 0118.48 9L11 16.53a1 1 0 01-.73.3z"
            fill="#172c42"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Setting up payroll between you and your caregiver.
        </p>
      </div>
      <div
        class="makeStyles-listItem-3"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-root-14 makeStyles-listIcon-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.27 16.83a1 1 0 01-.7-.3l-4.3-4.3a1 1 0 011.41-1.41l3.59 3.59 6.8-6.79A1 1 0 0118.48 9L11 16.53a1 1 0 01-.73.3z"
            fill="#172c42"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Ensuring your caregiver has the right tools to capture needed reimbursement information.
        </p>
      </div>
    </div>
    <div
      class="makeStyles-root-15"
    >
      <div
        class="makeStyles-content-16 makeStyles-information-20 makeStyles-roundCorners-18 makeStyles-fullWidth-17"
      >
        <div
          class="makeStyles-icon-19"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-root-25"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.28 20.77a13.16 13.16 0 01-5.77-2.64c-.64-.48-1.28-1-1.84-1.52l-.08-.08a17.09 17.09 0 01-1.52-1.77 12.52 12.52 0 01-2.88-5.6 4.69 4.69 0 01.32-3.45 6 6 0 013-2.32 1.61 1.61 0 01.88-.24 1.71 1.71 0 011.44.72A21 21 0 0110.56 6a2.07 2.07 0 01.08 2.33 3.26 3.26 0 01-.48.88 1.18 1.18 0 00.08 1.52c.48.56 1 1 1.52 1.6s1 .88 1.52 1.28a1.69 1.69 0 00.8.32 1.54 1.54 0 00.8-.24c.32-.24.56-.48.8-.64a2 2 0 012.32 0c.72.48 1.45 1 2.25 1.68a2 2 0 01.56 2.41A6 6 0 0119.13 20a4 4 0 01-2.65.88c-.4-.03-.8-.11-1.2-.11zM7 4.43C4.55 5.55 3.83 6.67 4.31 8.92A12.54 12.54 0 007 14a15.93 15.93 0 001.52 1.68c.56.48 1.12 1 1.77 1.45a11.49 11.49 0 005.2 2.4 3.89 3.89 0 001 .08c1.6 0 2.57-.8 3.37-2.88a.88.88 0 00-.24-1l-.08-.08L17.36 14a1 1 0 00-1 0 6.92 6.92 0 01-.8.56 2.6 2.6 0 01-3.12-.08l-1.64-1.4-.08-.08c-.48-.56-1.05-1-1.53-1.6a2.28 2.28 0 010-2.88 4.86 4.86 0 01.56-.8.85.85 0 00-.08-1 19.06 19.06 0 00-1.76-2.13.62.62 0 00-.56-.24 3.46 3.46 0 00-.35.08z"
              fill="#182c42"
              stroke="#182c42"
              stroke-miterlimit="10"
              stroke-width=".2"
            />
          </svg>
        </div>
        <div>
          <div>
            <h3
              class="MuiTypography-root MuiTypography-h3"
            >
              Be on the lookout for a call from us.
            </h3>
            <p
              class="MuiTypography-root makeStyles-wantToSpeakWithSomeone-8 MuiTypography-body2"
            >
              Want to speak with someone immediately? 
            </p>
          </div>
          <div
            class="makeStyles-phoneNumberContainer-7 makeStyles-phoneNumberContainer-10"
          >
            <p
              class="MuiTypography-root makeStyles-call-9 MuiTypography-body2"
            >
              Call
            </p>
            <a
              class="MuiTypography-root MuiLink-root makeStyles-root-26 MuiLink-underlineNone MuiTypography-colorPrimary"
              href="tel:877-367-1959"
            >
              <p
                class="MuiTypography-root makeStyles-link2-40 MuiTypography-body1"
              >
                877-367-1959
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
